### PR TITLE
added `change_objective!` function and tests

### PR DESCRIPTION
**Pull request by stelmo**
Added a small convenience function and tests

### CODE REVIEW

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit e2d3a7f9 is formatted properly.

File: src/reconstruction/CoreModel.jl [hunk 1]
```diff
@@ -436,3 +436,41 @@ end
...
+model, sets objective. Note, also accepts `String` or `Vector{String}` of reaction ids.
+"""
+function change_objective!(
+    model::CoreModel,
+    rxn_idxs::Vector{Int};
+    weights = ones(length(rxn_idxs)),
+)
+    nz_idxs, _ = findnz(objective(model))
+    model.c[nz_idxs] .= 0.0 # reset
+    dropzeros!(model.c) # cleanup
```
Comment (exaexa) [inline src/reconstruction/CoreModel.jl:458]: at this point it is better and faster to just do:
```julia
model.c=spzeros(length(model.c))
model.c[rxn_idxs] .= weigths
```

File: src/reconstruction/CoreModelCoupled.jl [hunk 2]
```diff
@@ -424,3 +424,36 @@ end
...
+
+Change the objective for `model` to reaction(s) with indices `rxn_ids`, optionally
+specifying their `weights`. By default, assume equal weights. If no objective exists in
+model, sets objective.
+"""
+function change_objective!(
+    model::CoreModelCoupled,
+    rxn_idxs::Vector{Int};
+    weights = ones(length(rxns)),
+)
```
Comment (exaexa) [inline src/reconstruction/CoreModelCoupled.jl:443]: forward parameters with `...` and refer to the implementing function (shorter, allows extensibility)

**Review by exaexa — CHANGES_REQUESTED** [review]

File: src/reconstruction/StandardModel.jl [hunk 3]
```diff
@@ -230,3 +230,33 @@ end
...
+function change_objective!(
+    model::StandardModel,
+    rxn_ids::Vector{String};
+    weights = ones(length(rxn_ids)),
+)
+    all(!haskey(model.reactions, rid) for rid in rxn_ids) &&
+        throw(DomainError(rxn_ids, "Some reaction ids were not found in model."))
+    k = 1 # counter
+    for rid in reactions(model)
+        if rid in rxn_ids
```
[the remark this answers is not in the document]
Comment (stelmo) [reply src/reconstruction/StandardModel.jl]: ah you are right, I was trying to avoid the double loop.

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 138a24f4 requires formatting!

Required formatting changes summary:
```
 src/reconstruction/CoreModelCoupled.jl | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)
```

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 0b8183ca requires formatting!

Required formatting changes summary:
```
 src/reconstruction/CoreModelCoupled.jl | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)
```

**Comment by cylon-x** [comment]
:red_square: &nbsp;Commit 9ee5e36f requires formatting!

Required formatting changes summary:
```
 src/reconstruction/CoreModelCoupled.jl | 12 ++----------
 1 file changed, 2 insertions(+), 10 deletions(-)
```

File: src/reconstruction/StandardModel.jl [hunk 4]
```diff
@@ -230,3 +230,31 @@ end
...
+    model::StandardModel,
+    rxn_ids::Vector{String};
+    weights = ones(length(rxn_ids)),
+)
+    all(!haskey(model.reactions, rid) for rid in rxn_ids) &&
+        throw(DomainError(rxn_ids, "Some reaction ids were not found in model."))
+
+    for rid in reactions(model) # reset to zero
+        model.reactions[rid].objective_coefficient = 0.0
+    end
```
Comment (exaexa) [inline src/reconstruction/StandardModel.jl:254]: avoid unnecessary indexing and explicit Float64s
```julia
for (_,r) in model.reactions
  r.o..._c... = 0
end
```

**Comment by cylon-x** [comment]
:green_circle: &nbsp;Commit 53d876e9 is formatted properly.

**Review by exaexa — APPROVED** [review]

**Pull request merged**